src/Reflection/ResolvedFunctionVariant.php

```diff
@@ -73,10 +81,29 @@ public function getReturnType(): Type
 				$this->resolvedTemplateTypeMap,
 			);
 
+			$type = $this->resolveConditionalTypes($type);
+
 			$this->returnType = $type;
 		}
 
 		return $type;
 	}
 
+	private function resolveConditionalTypes(Type $type): Type
+	{
+		return TypeTraverser::map($type, function (Type $type, callable $traverse): Type {
+			if ($type instanceof ConditionalTypeForParameter) {
+				if (array_key_exists($type->getParameterName(), $this->passedArgs)) {
+					$type = $type->toConditional($this->passedArgs[$type->getParameterName()]);
+				}
+			}
+
+			if ($type instanceof ConditionalType) {
+				$type = $type->resolve();
+			}
+
+			return $traverse($type);
```

Comment:
This looks like it supports nested conditionals but they're not tested :) Please add some test conditional type inside conditional type :) Thanks.